db/warehouse/migrate/20250522193546_add_data_source_id_to_wfd_template.rb

```diff
@@ -0,0 +1,32 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+class AddDataSourceIdToWfdTemplate < ActiveRecord::Migration[7.0]
+  def change
+    safety_assured do # not yet used in prod, so safety_assured is ok
+      add_column :wfd_templates, :data_source_id, :integer, null: true
+
+      add_index :wfd_templates, :data_source_id
```

Comment:
suggest using add_reference instead to add a foreign key:
```
add_reference :wfd_templates, :data_source, foreign_key: true, null: true
```
(this would replace the add_column and add_index)

Reply:
Sounds good. We will have to split this out into 2 separate migrations, though, which are then run separately in any environment that already includes any templates. (so QA only, I believe.) The trick with setting them to 0 temporarily doesn't work if it is a foreign key.

We could also remove the nullability constraint.

Reply:
Opened https://github.com/greenriver/hmis-warehouse/pull/5431 to follow up.